src/dependencies.rb

```diff
@@ -0,0 +1,110 @@
+#==============================================================================
+# Copyright (C) 2017 Stephen F. Norledge and Alces Software Ltd.
+#
+# This file/package is part of Alces Metalware.
+#
+# Alces Metalware is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation, either version 3 of
+# the License, or (at your option) any later version.
+#
+# Alces Metalware is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this package.  If not, see <http://www.gnu.org/licenses/>.
+#
+# For more information on the Alces Metalware, please visit:
+# https://github.com/alces-software/metalware
+#==============================================================================
+require 'exceptions'
+require 'constants'
+
+module Metalware
+  class Dependencies
+    def initialize(metal_config, command_input, dep_hash = {})
+      @config = metal_config
+      @dep_hash = dep_hash
+      @command = command_input
+    end
+
+    def enforce
+      @dep_hash.each { |dep, values|
+        next unless values # Rejects nil and false
+        values = [values] unless values.is_a?(Array)
+        values.each { |value| validate_dependency(dep, value) }
+      }
+    end
+
+    private
+
+    attr_reader :config, :command
+
+    def validate_dependency(dep, value)
+      return if send(:"validate_#{dep}") == "skip"
```

Comment:
I don't really like relying on a magic string being returned to determine if we don't need to run a particular validation. If we need this, we could have a `should_validate_#{dep}?` method as well as a `validate_#{dep}` method, with the former determining if the validation is needed while the latter just does the actual validation. That would make things clearer and give each method a single purpose.

With that said, I can't really see a need for conditional validations. I think we should just enforce the validations in all cases; otherwise we are making things more complex without a real benefit, as no one is really depending on the previous behaviour in any case.

When it comes time for the next release we can decide if this is a breaking change or not, but we shouldn't make our code over-complicated to avoid having to decide this (and risk more bugs due to the added complexity in the process).